src/services/sentry.ts

```diff
@@ -42,6 +44,12 @@ export const reportError = (err: unknown): void => {
   if (__DEV__) {
     console.error('Sentry reported error: ', err)
   } else {
-    Sentry.captureException(err)
+    AsyncStorage.isTrackingEnabled()
+      .then(isTrackingEnabled => {
+        if (isTrackingEnabled) {
+          Sentry.captureException(err)
+        }
+      })
+      .catch(() => Sentry.captureException(err))
```

Comment:
Not sure if it makes sense to capture the original exception if something goes wrong
```suggestion
      .catch(Sentry.captureException)
```